### ISSUE TITLE
React 17/18 support 

### ISSUE THREAD

**Issue by jeetiss** [diegomura/react-pdf#2119]
`@react-pdf/renderer` has 2 problems with react 18: 

- https://github.com/diegomura/react-pdf/issues/1878
- https://github.com/diegomura/react-pdf/issues/1779

The last one can be solved easily, but the first one is hard because `react-reconciler` depends on `react` too. (from this message: https://github.com/diegomura/react-pdf/issues/1062#issuecomment-813804883)

It should work if we bundle `react-reconciler` into `@react-pdf/renderer` as `react-dom` does.
so we can override any dependencies it should help but it can be dangerous 

if you have a better option on how to fix this issue, welcome
if you wanna `bump` goto #1878

**Comment by jeetiss** [comment]
@diegomura what do you think about including `react-reconciler` into `@react-pdf/renderer`

**Comment by jeetiss** [comment]
should be fixed in `v3.1.1`

**Comment by mdodge-ecgrow** [comment]
Now that this is fixed, and I am on v3.1.4, I should be able to remove this code from my package.json, correct?
```
"overrides": {
    "@react-pdf/renderer": {
      "react": "^18.0.0"
    }
  },
```
